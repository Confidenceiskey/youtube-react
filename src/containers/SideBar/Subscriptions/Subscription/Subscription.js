import React from 'react';
import { Icon, Image, Menu } from 'semantic-ui-react';
import './Subscription.scss';

const Subscription = ({ broadcasting, amountNewVideos, label }) => {
  let rightElement = null;

  if (broadcasting) {
    rightElement = <Icon name='signal' />;
  } else if (amountNewVideos) {
    rightElement = <span className='new-videos-count'>{amountNewVideos}</span>;
  }

  return (
    <Menu.Item>
      <div className='subscription'>
        <div>
          <Image src='http://via.placeholder.com/28x28' avatar />
          <span>{label}</span>
        </div>
        {rightElement}
      </div>
    </Menu.Item>
  );
}

export default Subscription;
