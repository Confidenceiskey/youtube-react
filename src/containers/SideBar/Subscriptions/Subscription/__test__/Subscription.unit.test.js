import React from 'react';
import { shallow } from 'enzyme';
import Subscription from '../Subscription';

describe('various cases for Subscription', () => {
  it('renders empty subscription', () => {
    const wrapper = (
      <Subscription />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders subscription that is broadcasting', () => {
    const wrapper = (
      <Subscription broadcasting label='freeCodeCamp' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders non-broadcasting subscription with new vids', () => {
    const wrapper = (
      <Subscription amountNewVideos={7} label='Red Bull TV' />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
