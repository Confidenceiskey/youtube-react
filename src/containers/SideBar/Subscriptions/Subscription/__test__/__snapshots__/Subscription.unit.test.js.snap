// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`various cases for Subscription renders empty subscription 1`] = `<Subscription />`;

exports[`various cases for Subscription renders non-broadcasting subscription with new vids 1`] = `
<Subscription
  amountNewVideos={7}
  label="Red Bull TV"
/>
`;

exports[`various cases for Subscription renders subscription that is broadcasting 1`] = `
<Subscription
  broadcasting={true}
  label="freeCodeCamp"
/>
`;
